lib/ansible/modules/crypto/openssl_cert.py

```diff
@@ -0,0 +1,996 @@
+    signature_algorithms:
+      - sha224WithRSAEncryption
+      - sha256WithRSAEncryption
+      - sha384WithRSAEncryption
+      - sha512WithRSAEncryption
+      - sha224WithECDSAEncryption
+      - sha256WithECDSAEncryption
+      - sha384WithECDSAEncryption
+      - sha512WithECDSAEncryption
+
+- name: Ensure that the existing certificate belongs to the specified private key
+  openssl_cert:
+    dest: /etc/ssl/crt/example.com.crt
+    privatekey: /etc/ssl/private/example.com.pem
+    provider: assertonly
+
+- name: Ensure that the existing certificate is still valid at the winter solstice 2017
+  openssl_cert:
+    dest: /etc/ssl/crt/example.com.crt
+    provider: assertonly
+    valid_at: 2017-12-21T16:28:00Z
+
+- name: Ensure that the existing certificate is still valid 2 weeks (1209600 seconds) from now
+  openssl_cert:
+    dest: /etc/ssl/crt/example.com.crt
+    provider: assertonly
+    valid_in: 1209600
+
+- name: Ensure that the existing certificate does not use Diffie-Hellman
+  openssl_cert:
+    dest: /etc/ssl/crt/example.com.crt
+    provider: assertonly
+    keyUsage_contains:
+      - key_agreement: False
+
+- name: Ensure that the existing certificate is only used for digital signatures and encrypting other keys
+  openssl_cert:
+    dest: /etc/ssl/crt/example.com.crt
+    provider: assertonly
+    keyUsage_only:
+      - digital_signature: True
+      - key_encipherment: True
+
+- name: Ensure that the existing certificate can be used for client authentication
+  openssl_cert:
+    dest: /etc/ssl/crt/example.com.crt
+    provider: assertonly
+    extendedKeyUsage_contains:
+      - clientAuth
+
+- name: Ensure that the existing certificate can only be used for client authentication and time stamping
+  openssl_cert:
+    dest: /etc/ssl/crt/example.com.crt
+    provider: assertonly
+    extendedKeyUsage_only:
+      - clientAuth
+      - 1.3.6.1.5.5.7.3.8
+
+- name: Ensure that the existing certificate has a certain domain in its subjectAltName
+  openssl_cert:
+    dest: /etc/ssl/crt/example.com.crt
+    provider: assertonly
+    subjectAltName_contains:
+      - www.example.com
+      - test.example.com
+'''
+
+
+RETURN = '''
+cert_path:
+    description: Path to the generated Certificate
+    returned: changed or success
+    type: string
+    sample: /etc/ssl/crt/www.ansible.com.crt
+'''
+
+
+import datetime
+from random import randint
+import re
+import os
+
+try:
+    from OpenSSL import crypto
+    from OpenSSL import __version__ as opensslversion
+except ImportError:
+    pyopenssl_found = False
+else:
+    pyopenssl_found = True
+
+try:
+    from cryptography import x509
+    from cryptography.hazmat.backends import default_backend
+    from OpenSSL import __version__ as cryptographyversion
+except ImportError:
+    cryptography_found = False
+else:
+    cryptography_found = True
+
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.pycompat24 import get_exception
```

Comment:
This mechanism has been dropped lately. The correct way of doing it is by following the pattern introduced here https://github.com/ansible/ansible/commit/ed15a0e9a7bf237786634cc5dd89d9448031c86f